src/voter/voting_round.rs

```diff
@@ -199,11 +200,50 @@ impl<H, N, E: Environment<H, N>> VotingRound<H, N, E> where
 		let post_state = self.votes.state();
 		self.notify(pre_state, post_state);
 
-		if self.votes.completable() {
-			Ok(Async::Ready(()))
-		} else {
-			Ok(Async::NotReady)
+		// early exit if the current round is not completable
+		if !self.votes.completable() {
+			return Ok(Async::NotReady);
 		}
+
+		// make sure that the previous round estimate has been finalized
+		let last_round_estimate_finalized = match last_round_state {
+			Some(RoundState {
+				estimate: Some((last_round_estimate, _)),
+				finalized: Some((last_round_finalized, _)),
+				..
+			}) => {
+				// either it was already finalized in the previous round
+				let finalized_in_last_round = self.env.is_equal_or_descendent_of(
```

Comment:
Can we get away with just using block numbers?

Reply:
Yeah, that should be fine. So we are checking `last_round.estimate_number <= finalized_number` before starting `next_round`.